contents/docs/plugins/build.md

```diff
@@ -124,51 +189,60 @@ async function processEvent(event, { config, cache }) {
 
-#### config
+> **Note:** Your plugin can use `processEvent` or `processEventBatch`. Currently, if both are present, `processEventBatch` will not run. 
 
-Gives you access to the PostHog config as described in `plugin.json` and configured via the PostHog interface.
+Here's the same example from above, except now using `processEventBatch`: 
 
-#### cache
-
-A way to store values that persist across `processEvent` calls. The values are stored in [Redis](https://redis.io/), an in-memory store.
+```js
+async function processEventBatch(events, { config, cache }) {
+    const counter = await cache.get('counter', 0)
+    cache.set('counter', counter + 1)
 
-Storing values is done via `cache.set`, which takes the following parameters:
+    for (let event of events) {
+      if (event.properties) {
+          event.properties['hello'] = 'world'
+          event.properties['bar'] = config.bar
+          event.properties['$counter'] = counter
+      }
+    }
 
-```
-key: string, 
-value: unknown
+    return events
+}
 ```
 
-And retrieving them uses `cache.get`, which takes:
+As you can see, since `events` is an array of events, we iterate over it to access every individual event. 
 
-```
-key: string, 
-defaultValue: unknown
-```
+> **Note:** As of right now, batch size is not yet configurable and still receives only one event (albeit in array form).
 
-> If you're unfamiliar with TypeScript, you can read about the `unknown` type on this [blog post by Microsoft](https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type)
+### Scheduled Tasks
 
-### global
+Plugins can also run scheduled tasks through the functions:
 
-Global is used for sharing functionality between `setupPlugin` and `processEvent`, since global scope does not work in the context of PostHog plugins. 
+- `runEveryMinute`
+- `runEveryHour`
+- `runEveryDay`
 
-### attachments
+These functions only take an object of type `PluginMeta` as a parameter and do not return anything.
 
-Attachments gives access to files uploaded by the user for config parameters of type `attachment`. An `attachment` has the following type definition:
+Example usage:
 
 ```js
-interface PluginAttachment {
-    content_type: string
-    file_name: string
-    contents: any
+async function runEveryMinute ({cache}) {
+    const cursor = await cache.get('last_fetched_cursor')
+    const response = await fetch(`http://my.api.com/get_stargazers?since=${cursor}`)
+    for (const { username } of response.stargazers) {
+        posthog.capture('someone starred you!', { username })
+    }
+    const last_cursor = response.stargazers.map(s => s.created_at).sort().reverse()[0] || cursor
+    await cache.set('last_fetched_cursor', last_cursor)
```

Comment:
Here's a cooler example:
```js
async function runEveryMinute({ config }) {
    const url = `https://api.github.com/repos/PostHog/posthog`
    const response = await fetch(url)
    const metrics = await response.json()

    posthog.capture('github metrics', {
        stars: metrics.stargazers_count,
        open_issues: metrics.open_issues_count,
        forks: metrics.forks_count,
        subscribers: metrics.subscribers_count
    })
}
```

The problem with the other one is that it's pseudocode. This one is real.